### PR TITLE
X86 fix

### PR DESCRIPTION
**Pull request by Dominion5254**
Fixes X86 health check and ffmpeg.
Change `mediasources` config to enum
Hide extraneous folders from library selections
Refactor `dockerfile` and `docker_entrypoint.sh`

### CODE REVIEW

File: docker_entrypoint.sh [hunk 1]
```diff
@@ -2,13 +2,36 @@
 
 set -e
 
+# Get the architecture of the host
+ARCH=$(uname -m)
```
Comment (chrisguida) [inline docker_entrypoint.sh:6]: Clever.

File: docker_entrypoint.sh [hunk 2]
```diff
@@ -32,4 +55,12 @@ else
...
 fi
 
-exec ./jellyfin/jellyfin --datadir /config --cachedir /cache --ffmpeg /usr/bin/ffmpeg
+# Run Jellyfin with appropriate options based on the architecture
+if [ "$ARCH" = "aarch64" ]; then
+  exec ./jellyfin/jellyfin --datadir /config --cachedir /cache --ffmpeg /usr/bin/ffmpeg
+elif [ "$ARCH" = "x86_64" ]; then
+  exec ./jellyfin/jellyfin --datadir /config --cachedir /cache --ffmpeg /usr/lib/jellyfin-ffmpeg/ffmpeg
+else
+  echo "Unsupported architecture: $ARCH"
```
Comment (chrisguida) [inline docker_entrypoint.sh:64]: my mac says `arm64` for `uname -a`. what systems did you test this build on?

File: docker_entrypoint.sh [hunk 3]
```diff
@@ -2,13 +2,36 @@
...
+ARCH=$(uname -m)
 
 while [ ! -f /jellyfin/main/start9/config.yaml ]; do
     sleep 1
 done
 
 echo "File /jellyfin/main/start9/config.yaml has been created."
 
+grep -q "filebrowser" /jellyfin/main/start9/config.yaml && FILEBROWSER=true || FILEBROWSER=false
+grep -q "nextcloud" /jellyfin/main/start9/config.yaml && NEXTCLOUD=true || NEXTCLOUD=false
```
Comment (chrisguida) [inline docker_entrypoint.sh:15]: This seems a bit brittle, as it is not robust against adding the word "filebrowser" or "nextcloud" anywhere in the config file. you should use `yq` to select the specific value that affects this behavior.

**Review by chrisguida — CHANGES_REQUESTED** [review]
see comments

File: docker_entrypoint.sh (hunk 2 above)
Comment (Dominion5254) [reply docker_entrypoint.sh:64]: Initially I had used `arm64` in this conditional, but when I ran the s9pk on my Pi testbox, the logs were showing the error "Unsupported architecture aarch64" - So I replaced `arm64` with `aarch64`.

My understanding was arm64 and aarch64 were functionally the same, but if I was mistaken in this understanding, I can update the conditional to include the appropriate `exec` command for `arm64` in addition to `aarch64`

Comment (chrisguida) [reply docker_entrypoint.sh:64]: I would just sidestep this issue entirely by building the x86 and arm builds the same way

File: Dockerfile.amd64
Comment (chrisguida) [inline Dockerfile.amd64:1]: 🚀

File: Makefile [hunk 4]
```diff
@@ -33,11 +33,11 @@ x86: docker-images/x86_64.tar scripts/embassy.js
 
 docker-images/aarch64.tar: Dockerfile docker_entrypoint.sh $(JELLYFIN_SRC)
 	mkdir -p docker-images
-	docker buildx build --tag start9/$(PKG_ID)/main:$(PKG_VERSION) --build-arg PLATFORM=arm64 --platform=linux/arm64 -o type=docker,dest=docker-images/aarch64.tar .
+	docker buildx build --tag start9/$(PKG_ID)/main:$(PKG_VERSION) --build-arg PLATFORM=arm64 --build-arg ARCH=arm64 --platform=linux/arm64 -o type=docker,dest=docker-images/aarch64.tar .
 
 docker-images/x86_64.tar: Dockerfile docker_entrypoint.sh $(JELLYFIN_SRC)
 	mkdir -p docker-images
-	docker buildx build --tag start9/$(PKG_ID)/main:$(PKG_VERSION) --build-arg PLATFORM=amd64 --platform=linux/amd64 -o type=docker,dest=docker-images/x86_64.tar .
+	docker buildx build --tag start9/$(PKG_ID)/main:$(PKG_VERSION) --build-arg PLATFORM=amd64 --build-arg ARCH=x64 --platform=linux/amd64 -o type=docker,dest=docker-images/x86_64.tar .
```
Comment (chrisguida) [inline Makefile:40]: note: ARCH for `jellyfin-web` build; PLATFORM for `yq`

File: manifest.yaml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 id: jellyfin
 title: "Jellyfin"
-version: 10.8.9.1
-release-notes: "Minor fixes"
+version: 10.8.9.2
+release-notes: "Minor fixes and refactoring"
```
Comment (chrisguida) [inline manifest.yaml:4]:
```suggestion
release-notes: "Initial release of Jellyfin for StartOS"
```

Comment (chrisguida) [reply manifest.yaml:4]: I'll leave this to your discretion, as I'm not sure what the recommended best practice is here

File: scripts/procedures/health.ts [hunk 6]
```diff
@@ -1,12 +1,3 @@
-// import { types as T, checkWebUrl, catchError } from "../deps.ts";
-
-// export const health: T.ExpectedExports.health = {
-//     // deno-lint-ignore require-await
-//     async "alive"(effects, duration) {
-//       return checkWebUrl("http://jellyfin.embassy:8096")(effects, 90000).catch(catchError(effects))
-//     },
-// };
-
```
Comment (chrisguida) [inline scripts/procedures/health.ts:9]: Delete all commented code, please

**Review by chrisguida — CHANGES_REQUESTED** [review]
See comments

File: Dockerfile.amd64
Comment (chrisguida) [reply Dockerfile.amd64:1]: Wait is this file deleted?

File: docker_entrypoint.sh (hunk 3 above)
Comment (Dominion5254) [reply docker_entrypoint.sh]: This has been updated to use `yq` as suggested in cec7cfaec76f322a451bb847e11201aa0456d614

File: manifest.yaml (hunk 5 above)
Comment (Dominion5254) [reply manifest.yaml]: Per our discussion, this has been updated to `initial release` in 1be02f14e9672158205a3fcb04490a4bbae3d993

File: instructions.md [hunk 7]
```diff
@@ -2,18 +2,28 @@
...
 
-* By default, Jellyfin will scan your files for metadata using The Movie Database (TMDb) and Open Media Database (OMDb). This metadata is then used to fetch media images and information (such as summary, cast, etc) which is then displayed within the Jellyfin web app or other [client](https://jellyfin.org/downloads/clients). While these features make for a great interface and user experience, they can be disabled by unchecking the corresponding boxes. Repeat the bullets under step 3 for as many media libraries as needed.
+    **IMPORTANT: IF FILES ARE NOT SHOWING UP IN ONE OF YOUR LIBRARIES, DOUBLE CHECK THAT YOU DON'T HAVE ANY LIBRARIES WHOSE PATHS OVERLAP WITH OTHER NESTED LIBRARIES i.e. `/mnt/filebrowser` and `/mnt/filebrowser/videos`**
```
Comment (chrisguida) [inline instructions.md:19]: ✅

**Review by chrisguida — APPROVED** [review]
Ship it

**Pull request merged**